### PR TITLE
New Rule (Attachment): Archive contains DLL-loading macro

### PR DESCRIPTION
**Pull request by aidenmitchell**
Some work to be done:
- [x] Detect other archives, such as .iso
- [x] Detect the specific DLL-loading command in the macro script

### CODE REVIEW

**Comment by aidenmitchell** [comment]
@morriscode when you have a moment, could you review my syntax?

ISO https://analyzer.sublime.security?id=b8a5cdd7-8776-450a-872d-00bd82ae607f
ZIP: https://analyzer.sublime.security?id=81b0d441-9002-44f9-b60c-aa0aa1dacaf3
ZIP (encrypted): https://analyzer.sublime.security?id=f39d2b9d-71c3-490c-8940-e50073c03854

File: detection-rules/attachment_macro_dll_loader.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+     )
+  or any(attachments, .file_extension in~ $file_extensions_common_archives
+      and any(file.explode(.),
+          any(.flavors.yara, strings.like(., "vb_file"))
+          and any(.scan.strings.strings, strings.ilike(., "*Lib*.dll*"))
+      )
+      and any(file.explode(.),
+          strings.ilike(.file_extension, "dll")
+      )
+  )
```
Comment (morriscode) [inline detection-rules/attachment_macro_dll_loader.yml:36]:
```suggestion
         )
         or any(file.explode(.), // fallback for encrypted zips
             .scan.zip.encrypted == true
             and any(.scan.zip.all_paths, any($file_extensions_macros, strings.icontains(.., .)))
             // zip contains a dll file
             and any(.scan.zip.all_paths, strings.icontains(., "dll"))
         )
  or any(attachments, .file_extension in~ $file_extensions_common_archives
      and any(file.explode(.),
          any(.flavors.yara, strings.like(., "vb_file"))
          and any(.scan.strings.strings, strings.ilike(., "*Lib*.dll*"))
      )
      and any(file.explode(.),
          strings.ilike(.file_extension, "dll")
              )
      )
  )
```

**Comment by morriscode** [comment]
Been running in test rules for 2 weeks with no hits. 🚢  it

**Review by morriscode — APPROVED** [review]

**Pull request merged**